src/mapgen_carpathian.cpp

```diff
@@ -382,6 +386,9 @@ int MapgenCarpathian::generateTerrain()
 	noise_step_mnt->perlinMap2D(node_min.X, node_min.Z);
 	noise_mnt_var->perlinMap3D(node_min.X, node_min.Y - 1, node_min.Z);
 
+	// Grad at water_level (calculate once outside loops as a constant)
+	s32 grad_wl = terrain_zero_level - water_level;
```

Comment:
Instead of calculating this every mapchunk, this can be a class variable calculated in the constructor.